src/main/java/appeng/tile/inventory/AppEngInternalAEInventory.java

```diff
@@ -150,42 +141,85 @@ public ItemStack getStackInSlot( final int var1 )
 	}
 
 	@Override
-	public ItemStack decrStackSize( final int slot, final int qty )
+	public ItemStack insertItem( int slot, ItemStack stack, boolean simulate ) 
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The Cyclomatic Complexity of this method "insertItem" is 13 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)